presidio-anonymizer/presidio_anonymizer/entities/anonymizer_request.py

```diff
@@ -4,114 +4,56 @@
-        """
-        Get the right anonymizer_dto from the list.
-
-        When anonymizer_dto does not exist, we fall back to default.
-        :param analyzer_result: the result we are going to do the anonymization on
-        :return: anonymizer_dto
-        """
-        anonymizer_dto = self._anonymizers.get(entity_type)
-        if not anonymizer_dto:
-            anonymizer_dto = self._anonymizers.get("DEFAULT")
-            if not anonymizer_dto:
-                anonymizer_dto = self.default_anonymizer
-        anonymizer_dto["entity_type"] = entity_type
-        return anonymizer_dto
-
-    def get_text(self):
-        """Get the text we are working on."""
-        return self._text
-
-    def get_analysis_results(self):
-        """Get the analysis results."""
-        return self._analysis_results
-
-    def __validate_and_insert_input(self, data: dict):
-        self.__handle_text(data)
-        self.__handle_analyzer_results(data)
-        self.__handle_anonymizers(data)
-
-    def __handle_analyzer_results(self, data):
+    @classmethod
+    def handle_analyzer_results_json(cls, data) -> List[AnalyzerResult]:
         """
         Go over analyzer results, check they are valid and convert to AnalyzeResult.
 
-        :param data: contains the text, anonymizers and analyzer_results
+        :param data: contains the anonymizers and analyzer_results_json
         :return: None
         """
-        analyzer_results = data.get("analyzer_results")
-        if not analyzer_results:
-            self.logger.debug("invalid input, json missing field: analyzer_results")
+        analyzer_results = AnalyzerResults()
+        analyzer_results_json = data.get("analyzer_results")
+        if not analyzer_results_json:
+            cls.logger.debug(
+                "invalid input, json missing field: analyzer_results_json")
             raise InvalidParamException(
                 "Invalid input, " "analyzer results can not be empty"
             )
-        text_len = len(data.get("text"))
-        for analyzer_result in analyzer_results:
-            analyzer_result = AnalyzerResult(analyzer_result)
-            analyzer_result.validate_position_in_text(text_len)
-            self._analysis_results.append(analyzer_result)
+        for analyzer_result in analyzer_results_json:
+            analyzer_result = AnalyzerResult.from_json(analyzer_result)
+            analyzer_results.append(analyzer_result)
+        return analyzer_results
 
-    def __handle_anonymizers(self, data):
+    @classmethod
+    def get_anonymizer_configs_from_json(cls, data) -> Dict[str, AnonymizerConfig]:
         """
         Go over the anonymizers and get the relevant anonymizer class for it.
 
         Inserts the class to the anonymizer so the engine will use it.
         :param data: contains the text, anonymizers and analyzer_results
-        :return: None
+        :return: dict: key - entity_type of the anonymizer
+        value - AnonynmizerConfig
```

Comment:
No need for specifying the return, the type hint does it.

Reply:
Done.